### PR TITLE
feat: De-serializer generator part 1

### PR DESCRIPTION
**Pull request by jjtang1985**
This PR updates the generator for new clients, including:
- entity
- request builder
- entity api
- complex type

The missing parts are:
- batch
- function import
- nested complex type (tests)
- collection type (odata v4)
- enum type (odata v4)

Please note, the transpilation still fails without fixing the missing part mentioned.

### CODE REVIEW

**Comment by jjtang1985** [comment]
@marikaner , please feel free to merge.

File: packages/generator/src/generator-without-ts-morph/entity-api/class.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  ['Int32T', 'number'],
+  ['Int64T', 'BigNumber'],
+  ['GuidT', 'string'],
+  ['SByteT', 'number'],
+  ['SingleT', 'number'],
+  ['StringT', 'string'],
+  ['AnyT', 'any'],
+  ['DateTimeT', 'moment.Moment'],
+  ['DateTimeOffsetT', 'moment.Moment'],
+  ['TimeT', 'Time']
```
Comment (marikaner) [inline packages/generator/src/generator-without-ts-morph/entity-api/class.ts:66]: This is only correct for OData v2, but I don't see the notion of v2 in here. We have to differentiate this for v2 and v4.

File: packages/generator/src/generator-without-ts-morph/entity-api/file.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      typeOnly: false
+    },
+    {
+      names: [`${entity.className}RequestBuilder`],
+      moduleIdentifier: `./${entity.className}RequestBuilder`,
+      typeOnly: false
+    },
+    ...externalImports(entity.properties),
+    ...complexTypeImports(entity.properties),
+    // todo test odata v4
```
Comment (marikaner) [inline packages/generator/src/generator-without-ts-morph/entity-api/file.ts:32]: Don't forget the TODO.

**Review by marikaner — APPROVED** [review]
See my comments.

**Pull request merged**